runner/src/main.rs

```diff
@@ -176,6 +177,67 @@ fn run_tests_tsan() -> Step {
     }
 }
 
+pub fn run_fuzz_targets(opt: &RunFuzzTargets) -> Step {
+    let cargo_manifests: Vec<(PathBuf, CargoManifest)> = crate_manifest_files()
+        .filter(|path| is_fuzzing_toml_file(path))
+        .filter(|path| match &opt.crate_name {
+            Some(crate_name) => {
+                let mut crate_path = path.clone();
+                crate_path.pop();
+                crate_path.pop();
+                crate_path.file_name().and_then(|s| s.to_str()).unwrap() == crate_name.as_str()
+            }
+            None => true,
+        })
+        .map(|path| {
+            let cargo_manifest = toml::from_str(&read_file(&path)).unwrap_or_else(|err| {
+                panic!("could not parse cargo manifest file {:?}: {}", path, err)
+            });
+            (path, cargo_manifest)
+        })
+        .collect();
+
+    Step::Multiple {
+        name: "fuzzing".to_string(),
+        steps: cargo_manifests
+            .iter()
+            .map(|(path, cargo_manifest)| run_fuzz_targets_in_crate(path, &cargo_manifest, opt))
+            .collect(),
+    }
+}
+
+pub fn run_fuzz_targets_in_crate(
+    path: &PathBuf,
+    cargo_manifest: &CargoManifest,
+    opt: &RunFuzzTargets,
+) -> Step {
+    // `cargo-fuzz` can only run in the crate that contains the `fuzz` crate. So we need to use
+    // `Cmd::new_in_dir` to execute the command inside the crate's directory.
```

Comment:
But why is the code popping two elements from the path?

Reply:
Added an explanation.